confgenerator/confgenerator_test.go

```diff
@@ -181,17 +182,29 @@ func updateOrCompareGolden(t *testing.T, testName string, goos string, expectedB
 	expected := strings.ReplaceAll(string(expectedBytes), "\r\n", "\n")
 	actual = strings.ReplaceAll(actual, "\r\n", "\n")
 	goldenPath := fmt.Sprintf(path, goos, testName)
-	if diff := cmp.Diff(expected, actual); diff != "" {
-		if *updateGolden {
+	diff := cmp.Diff(expected, actual)
+	if *updateGolden {
+		if diff != "" || !fileExists(t, testName, goldenPath) {
```

Comment:
Now that line 170 checks for `os.ErrNotExist`, you can technically save yourself the `os.Stat` call, since you'd only have `diff == "" && actual == ""` (or `diff == "" && expected == ""`) when the file does not exist or was empty in the first place. The only difference between those is whether you'd want to overwrite (i.e., change the last modification time of) the empty file in the second case. With git, it usually doesn't matter, so we could go for the cheaper (and clearer) alternative. WDYT?

Reply:
True I guess if we're okay to make the tradeoff of overwriting the files when nothing needs to be written, then that would be clearer. The downside would be that if someone's doing a `-v` run they'll see golden files "changing" when they're not (the same empty files are just being overwritten). We would probably need to add a clarification for that somewhere if we did that. I might just be concerned over nothing though, and it would look clearer in the code.

Reply:
I've gone ahead and made that change, and if anything changes I can revert it or I can try to figure out where the clarification should go.

Reply:
It might be worth adding a comment, e.g. `// When actual is empty, the file may not exist; overwrite it`… WDYT?

Reply:
I added a clarifying comment.